src/libs/antares/study/parts/short-term-storage/makeGroupsOfHoursFromString.cpp

```diff
@@ -0,0 +1,130 @@
+
+#include <regex>
+
+#include <boost/algorithm/string.hpp>
+
+namespace Antares::Data::ShortTermStorage
+{
+
+static void throwExceptionWithMsg(const bool doWeThrow, const std::string& msg)
+{
+    if (doWeThrow)
+    {
+        throw std::invalid_argument(msg);
+    }
+}
+
+static bool onlyCommasOutsideBrackets(const std::string& hoursField)
+{
+    const std::regex strange_char_outside_brackets(R"(\][^\[\]]*[^,][^\[\]]*\[)");
+    return !std::regex_search(hoursField, strange_char_outside_brackets);
+}
+
+static bool oneCommaOutsideBrackets(const std::string& hoursField)
+{
+    const std::regex two_or_more_commas_outside_brackets(R"(\][,]{2,}\[)");
+    return !std::regex_search(hoursField, two_or_more_commas_outside_brackets);
+}
+
+static bool twoBracketsOpened(const std::string& hoursField)
+{
+    const std::regex twoBracketsOpened(R"(\[[^\[\]]*\[)");
+    return std::regex_search(hoursField, twoBracketsOpened);
+}
+
+static bool twoBracketsClosed(const std::string& hoursField)
+{
+    const std::regex twoBracketsClosed(R"(\][^\[\]]*\])");
+    return std::regex_search(hoursField, twoBracketsClosed);
+}
+
+static void checkNothingFancyOutsideBrackets(const std::string& hoursField)
+{
+    // Outside square brackets, we want only 1 comma, for ex : : "...] , [..."
+    throwExceptionWithMsg(!onlyCommasOutsideBrackets(hoursField), "strange char outside brackets");
+    throwExceptionWithMsg(!oneCommaOutsideBrackets(hoursField), "multiple commas outside brackets");
+}
+
+static void checkNoNestedSquareBrackets(const std::string& hoursField)
+{
+    throwExceptionWithMsg(twoBracketsOpened(hoursField), "2 square brackets opened");
+    throwExceptionWithMsg(twoBracketsClosed(hoursField), "2 square brackets closed");
+}
+
+static std::vector<std::string> splitIntoGroups(const std::string& hoursField)
+{
+    std::vector<std::string> my_groups;
+
+    std::regex betweenSquareBrackets(R"(\[([^\[\]]*?)\])");
+    auto matchBegin = std::sregex_iterator(hoursField.begin(),
+                                           hoursField.end(),
+                                           betweenSquareBrackets);
+    auto matchEnd = std::sregex_iterator();
+    for (auto it = matchBegin; it != matchEnd; ++it)
+    {
+        std::string match = (*it)[1].str();
+        my_groups.push_back(match);
+    }
+
+    return my_groups;
+}
+
+static std::vector<std::vector<std::string>> splitGroupsIntoHoursAsString(
+  const std::vector<std::string>& groups)
+{
+    std::vector<std::vector<std::string>> to_return;
+
+    std::regex commaSeparatedDigits(R"(^\d+(,\d+)*$)");
+    for (const auto& group: groups)
+    {
+        if (!std::regex_match(group, commaSeparatedDigits))
+        {
+            throw std::invalid_argument("splitting each group into hours > wrong format");
+        }
+        std::vector<std::string> hoursAsStrings;
+        boost::split(hoursAsStrings, group, boost::is_any_of(","));
```

Comment:
`boost::split` is slower than `std::getline`

Reply:
Ok.
How do you know that ?